From eb05a0762c5829652cdead0559f22c7102a1aa54 Mon Sep 17 00:00:00 2001
From: Vicki Pfau <vi@endrift.com>
Date: Wed, 14 Feb 2018 19:12:38 -0800
Subject: [PATCH] Disable pthread on Windows

---
 build/make/configure.sh | 2 +-
 configure               | 7 ++++++-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/build/make/configure.sh b/build/make/configure.sh
index 007e02000..2feba1f75 100644
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -1393,7 +1393,7 @@ EOF
   # Almost every platform uses pthreads.
   if enabled multithread; then
     case ${toolchain} in
-      *-win*-vs*)
+      *-win*-*)
         ;;
       *-android-gcc)
         # bionic includes basic pthread functionality, obviating -lpthread.
diff --git a/configure b/configure
index fb732acf3..e0c099044 100755
--- a/configure
+++ b/configure
@@ -554,7 +554,12 @@ process_detect() {
 int main(void) {return 0;}
 EOF
     # check system headers
-    check_header pthread.h
+    case $toolchain in
+        *-win*-*)
+            ;;
+        *)
+            check_header pthread.h
+    esac
     check_header unistd.h # for sysconf(3) and friends.
 
     check_header vpx/vpx_integer.h -I${source_path} && enable_feature vpx_ports
-- 
2.16.1

