diff --git a/build/make/configure.sh b/build/make/configure.sh
index 00214c9d2..cd199e18b 100644
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -1541,7 +1541,7 @@ EOF
   # Almost every platform uses pthreads.
   if enabled multithread; then
     case ${toolchain} in
-      *-win*-vs*)
+      *-win*-*)
         ;;
       *-android-gcc)
         # bionic includes basic pthread functionality, obviating -lpthread.
diff --git a/configure b/configure
index 6204f10b6..737634779 100755
--- a/configure
+++ b/configure
@@ -582,11 +582,17 @@ EOF
 
     # Use both check_header and check_lib here, since check_lib
     # could be a stub that always returns true.
-    check_header pthread.h && check_lib -lpthread <<EOF || disable_feature pthread_h
+    case $toolchain in
+        *-win*-*)
+            ;;
+        *)
+            check_header pthread.h && check_lib -lpthread <<EOF || disable_feature pthread_h
 #include <pthread.h>
 #include <stddef.h>
 int main(void) { return pthread_create(NULL, NULL, NULL, NULL); }
 EOF
+            ;;
+    esac
     check_header unistd.h # for sysconf(3) and friends.
 
     check_header vpx/vpx_integer.h -I${source_path} && enable_feature vpx_ports
